### PR TITLE
API Keys page redesign & descriptions for API connections

### LINKED ISSUE #838: Add information for what functionality API keys are needed.
It is not clear what keys are needed for the application to work fully, or what features will not work without.

Screen Shot 2020-04-04 at 14 28 17

### PR DESCRIPTION
**Pull request by isidorosp**
Fixes #838

* Added rotki.readthedocs.io to the URL whitelist (background.ts)
* Created BaseExternalLink component that can be used in both electron and web-version to create links outside the rotki "app" (BaseExteranLink.vue)
* Added tabs to the "API Keys" page (Rotki Premium / Exchanges / External Services), with descriptions and external links, where appropriate, for each tab (ApiKeys.vue, ExchangeSettings.vue, ExternalServices.vue, PremiumSettings.vue)
* Added the ability to provide a description for the cards info on each external service (ServiceKey.vue)
* Changed the external link in Statistics to use the BaseExternalLink component (Statistics.vue)
* Changelog entry for above

[ui tests]

### CODE REVIEW

**Comment by kelsos** [comment]
@isidorosp quick question did you run the integration tests? this should also run with `[ui tests]` flag

**Comment by isidorosp** [comment]
@kelsos What can I do in this PR to get Actions to also run them to confirm they're fixed?

**Comment by kelsos** [comment]
add [ui tests] to the descriptions maybe a separate row because actions is a bit picky

File: electron-app/src/views/settings/ApiKeys.vue [hunk 1]
```diff
@@ -5,9 +5,20 @@
         <h1>API Keys</h1>
       </v-col>
     </v-row>
-    <premium-settings></premium-settings>
-    <exchange-settings></exchange-settings>
-    <external-services></external-services>
+    <v-row>
+      <v-col>
+        Rotki can communciate with various APIs to provide additional
```
Comment (LefterisJP) [inline electron-app/src/views/settings/ApiKeys.vue:10]: communciate -> communicate

File: electron-app/src/components/settings/PremiumSettings.vue [hunk 2]
```diff
@@ -2,16 +2,25 @@
   <v-row class="premium-settings">
     <v-col>
       <v-card>
-        <v-card-title>Premium Settings</v-card-title>
+        <v-card-title>Rotki Premium</v-card-title>
         <v-card-text>
+          <p>
+            Rotki Premium is an optional subscription service to gain access to
```
Comment (LefterisJP) [inline electron-app/src/components/settings/PremiumSettings.vue:8]: to gain access to ...  
->
analytics, graphs and unlock many additional features. For more information on what is available visit the Rotki premium website.

File: electron-app/src/components/settings/api-keys/ExchangeSettings.vue [hunk 3]
```diff
@@ -2,8 +2,20 @@
   <v-row class="exchange-settings">
     <v-col>
       <v-card>
-        <v-card-title>Exchange Settings</v-card-title>
+        <v-card-title>Exchanges</v-card-title>
         <v-card-text>
+          <p>
+            Rotki can connect to supported exchanges and automatically obtain
```
Comment (LefterisJP) [inline electron-app/src/components/settings/api-keys/ExchangeSettings.vue:8]: obtain -> pull

File: electron-app/src/components/settings/api-keys/ExchangeSettings.vue [hunk 4]
```diff
@@ -2,8 +2,20 @@
...
+            your trades and balances in that exchange. Each exchange may require
```
Comment (LefterisJP) [inline electron-app/src/components/settings/api-keys/ExchangeSettings.vue:9]: in that exchange -> from that exchange

File: electron-app/src/components/settings/api-keys/ExchangeSettings.vue [hunk 5]
```diff
@@ -2,8 +2,20 @@
...
+            a different set of credentials for the connection, so make sure that
```
Comment (LefterisJP) [inline electron-app/src/components/settings/api-keys/ExchangeSettings.vue:10]: What do you want to say by:
"Each exchange may require a different set of credentials for the connection, so make sure that everything is provided"?

File: electron-app/src/components/settings/api-keys/ExternalServices.vue [hunk 6]
```diff
@@ -1,8 +1,18 @@
...
-  <div>
+  <v-container>
     <v-row>
       <v-col cols="12">
-        <h2>External Services</h2>
+        <v-card>
+          <v-card-title>External Services</v-card-title>
+          <v-card-text>
+            <p>
+              Rotki can connect to service providers in order to obtain more
```
Comment (LefterisJP) [inline electron-app/src/components/settings/api-keys/ExternalServices.vue:9]: I would rephrase to:

Rotki connects to various service providers in order to obtain information such as historical prices or blockchain data. In certain cases Rotki depends on these APIs for basic information, in which case you will need to provide an API key.

File: electron-app/src/components/settings/api-keys/ExternalServices.vue [hunk 7]
```diff
@@ -11,6 +21,7 @@
           v-model="etherscanKey"
           class="external-services__etherscan-key"
           title="Etherscan"
+          description="Required for any Ethereum blockchain balances or transactions. Rotki uses etherscan to obtain basic information about ethereum blockchain addresses and transactions."
```
Comment (LefterisJP) [inline electron-app/src/components/settings/api-keys/ExternalServices.vue:24]: Required for any Ethereum blockchain balances or transactions. Rotki uses etherscan to query ethereum blockchain data if you are not using your own ethereum node. If you are it's still needed but only for historical data.

File: electron-app/src/components/settings/api-keys/ExternalServices.vue [hunk 8]
```diff
@@ -41,6 +53,7 @@
           v-model="alethioKey"
           class="external-services__alethio-key"
           title="Alethio"
+          description="Rotki uses Alethio for supplementary Ethereum blockchain information. An API key is only needed if you have a lot of assets and are being rate-limited."
```
Comment (LefterisJP) [inline electron-app/src/components/settings/api-keys/ExternalServices.vue:56]: > for supplementary Ethereum blockchain information
change to: "to query Ethereum blockchain information".

File: electron-app/src/components/settings/api-keys/ExchangeSettings.vue (hunk 5 above)
Comment (isidorosp) [reply electron-app/src/components/settings/api-keys/ExchangeSettings.vue:10]: Some require more than just API Key + Secret (well it's only one right now), but perhaps it's redundant if we're linking to the guide.

File: electron-app/src/components/settings/PremiumSettings.vue [hunk 9]
```diff
@@ -2,16 +2,25 @@
...
+            analytics, graphs, and unlock many additional features. For more
+            information on what is available visit the
+            <base-external-link href="https://rotki.com/products/">
```
Comment (kelsos) [inline electron-app/src/components/settings/PremiumSettings.vue:11]: `ElectronIntrop` (probably bad choice for a name we should change it) should have the `premiumUrl` property why not use that instead of hardcoding it? It would be better to have the links in one place in geenral

File: electron-app/src/views/Statistics.vue [hunk 10]
```diff
@@ -10,7 +10,9 @@
         No premium subscription detected. Statistics are only available to
         premium users. <br />
         To get a premium subscription please visit our
-        <a :href="$interop.premiumURL" target="_blank">website</a>.
+        <base-external-link href="https://rotki.com/products/">
```
Comment (kelsos) [inline electron-app/src/views/Statistics.vue:13]: Same here no need to repeat the same URL everywhere

File: electron-app/src/components/settings/PremiumSettings.vue (hunk 9 above)
Comment (isidorosp) [reply electron-app/src/components/settings/PremiumSettings.vue:11]: I'm hard-coding it because if there is no $interop then it won't render the href properly (i.e. if the app is being run from the browser but no electron).

File: electron-app/src/views/Statistics.vue (hunk 10 above)
Comment (isidorosp) [reply electron-app/src/views/Statistics.vue:13]: See above.

File: electron-app/src/components/settings/PremiumSettings.vue (hunk 9 above)
Comment (kelsos) [reply electron-app/src/components/settings/PremiumSettings.vue:11]: @isidorosp yes that is why I mentioned that the name is a bad choice. This class is available in both platforms any suggestions for a suitable name?

Comment (isidorosp) [reply electron-app/src/components/settings/PremiumSettings.vue:11]: Sorry I didn't realize that interop is available even when we're not in electron. In that case we need to amend the code in https://github.com/rotki/rotki/blob/3d2ae3604bb576576c6a1f38a80caaffb1f3f5b4/electron-app/src/components/base/BaseExternalLink.vue because it just checks if $interop exist in order to determine whether to render a normal <a href="..."> link or to use an onclick event (if we're in electron).

Comment (isidorosp) [reply electron-app/src/components/settings/PremiumSettings.vue]: Changed it to `$interop.isPackaged` like you had originally suggested (I confirmed it works in dev as well).
